### PR TITLE
[flaky] Snapshot test case - improve test reports and cleanup

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:

Use Satisfy matcher to simplify the code and improve the failure message. Add correct offset to failure message.
Additionally improved the report and the namespace cleanup (might help with the Aftersuite failures)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
Just re-runing storage tests, maybe I can catch the failure

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -239,6 +240,15 @@ func (r *KubernetesReporter) logVMIMs(virtCli kubecli.KubevirtClient, vmims *v12
 	r.logObjects(virtCli, vmims, "vmims")
 }
 
+func (r *KubernetesReporter) logVMSnapshot(virtCli kubecli.KubevirtClient) {
+	snapshots, err := virtCli.VirtualMachineSnapshot(v1.NamespaceAll).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch vms: %v\n", err)
```
Comment (ShellyKa13) [inline tests/reporter/kubernetes.go:246]: the print should probably be failed to fetch vm snapshots

File: tests/testsuite/namespace.go [hunk 2]
```diff
@@ -209,6 +209,33 @@ func CleanNamespaces() {
...
+				}
+			}
+		}
+
+		util.PanicOnError(virtCli.VirtualMachineSnapshotContent(namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{}))
+		snapshotContentList, err := virtCli.VirtualMachineSnapshotContent(namespace).List(context.Background(), metav1.ListOptions{})
+		util.PanicOnError(err)
+		vmSnapshotContentFinalizer := "snapshot.kubevirt.io/vmsnapshotcontent-protection"
+		for _, snapshotContent := range snapshotContentList.Items {
+			if controller.HasFinalizer(&snapshotContent, vmSnapshotContentFinalizer) {
```
Comment (ShellyKa13) [inline tests/testsuite/namespace.go:232]: regarding this I wander if this wont cover a bug, supposedly removing the finalizer from vmsnapshotcontent is the responsibility of the vmsnapshot controller so if that is not dont maybe there is an actual issue

Comment (brybacki) [reply tests/testsuite/namespace.go:232]: but this is the `Cleanup` where we want to cleanup everything, so next tests could be run in clean environment, if you want to check for finalizer removal this is better done explicitly in some test.

File: tests/reporter/kubernetes.go (hunk 1 above)
Comment (brybacki) [reply tests/reporter/kubernetes.go:246]: yes, it should!

File: tests/reporter/kubernetes.go [hunk 3]
```diff
@@ -164,6 +164,7 @@ func (r *KubernetesReporter) dumpNamespaces(duration time.Duration, vmiNamespace
 	r.logNodes(virtCli, nodes)
 	r.logPods(virtCli, pods)
 	r.logVMs(virtCli)
+	r.logVMSnapshot(virtCli)
```
Comment (ShellyKa13) [inline tests/reporter/kubernetes.go:167]: it will be great if you can also add it for vmrestore

File: tests/reporter/kubernetes.go (hunk 1 above)
Comment (brybacki) [reply tests/reporter/kubernetes.go]: done

File: tests/reporter/kubernetes.go (hunk 3 above)
Comment (brybacki) [reply tests/reporter/kubernetes.go:167]: done

**Comment by brybacki** [comment]
The flakiness was fixed by: https://github.com/kubevirt/kubevirt/pull/8539

Anyway this PR is still needed and provides a few useful additions (cleanup/reporting)

**Pull request merged**